pci/src/vfio.rs

```diff
@@ -886,11 +908,26 @@ impl VfioCommon {
                     self.vfio_wrapper
                         .enable_irq(VFIO_PCI_INTX_IRQ_INDEX, vec![&eventfd])
                         .map_err(VfioPciError::EnableIntx)?;
-
-                    intx.enabled = true;
                 } else {
                     return Err(VfioPciError::MissingNotifier);
                 }
+
+                #[cfg(target_arch = "x86_64")]
+                {
+                    intx.interrupt_source_group
+                        .enable_resamplefd(0)
+                        .map_err(VfioPciError::EnableResamplefd)?;
+
+                    if let Some(reventfd) = intx.interrupt_source_group.unmask_notifier(0) {
+                        self.vfio_wrapper
+                            .set_irq_resample_fd(VFIO_PCI_INTX_IRQ_INDEX, vec![&reventfd])
+                            .map_err(VfioPciError::EnableIntx)?;
+                    } else {
+                        return Err(VfioPciError::MissingResampleNotifier);
+                    }
+                }
```

Comment:
Do we need to do the counter part for `disable_intx()`?

Reply:
The same as the above question. I will add the counter part.

Reply:
> The same as the above question. I will add the counter part.

@rbradford already done